lib/facter.rb

```diff
@@ -410,7 +410,7 @@ def trace(bool)
     # @api public
     def value(user_query)
       user_query = user_query.to_s.downcase
-      resolve_fact(user_query)
+      resolve_fact(user_query) unless @already_searched[user_query]&.value
```

Comment:
ah one gotcha, if a fact value is `false`, then thjis would resolve it multiple times. Recommend checking if the key is in the hash:

```ruby
... unless @already_searched.include?(user_query)
```